scripts/utils.sh

```diff
@@ -4,13 +4,17 @@ function die() {
   endc='\033[0m'
 
   if [[ $# -ge 1 ]]; then
-    echo -e "\n${red}>>> $@ <<<${endc}\n"
+    if [[ -t 0 ]]; then
+      echo -e "\n${red}>>> $@ <<<${endc}\n"
+    else
+      echo -e "\n>>> $@ <<<\n"
+    fi
   fi
   exit 1
 }
 
 function extract_version_from_gradle_file_stdin() {
-  tty --silent && die "${FUNCNAME[0]}: expecting non-terminal stdin, aborting" || true
+  [ -t 0 ] && die "${FUNCNAME[0]}: expecting non-terminal stdin, aborting" || true
```

Comment:
Maybe `[[ ... ]]` for consistency (actually, `[` is a part of POSIX but less feature-rich, `[[` is bash/zsh/...-specific and more feature-rich; for `-t` it doesn't matter anyway)

Reply:
Nowadays, using `[` mostly makes sense in ultralight Docker containers where no shell beyond `sh` satisfying the mere POSIX minimum is available; typically, those containers are some `alpine` derivations (https://hub.docker.com/_/alpine).

Reply:
good point!